app/Main.hs

```diff
@@ -82,22 +84,17 @@ formatOne ConfigFileOpts {..} mode reqSourceType rawConfig mpath =
   withPrettyOrmoluExceptions (cfgColorMode rawConfig) $ do
     let getCabalInfoForSourceFile' sourceFile = do
           cabalSearchResult <- getCabalInfoForSourceFile sourceFile
-          let debugEnabled = cfgDebug rawConfig
           case cabalSearchResult of
             CabalNotFound -> do
-              when debugEnabled $
-                hPutStrLn stderr $
-                  "Could not find a .cabal file for " <> sourceFile
+              logDebugM "CABAL FILE" $ "Could not find a .cabal file for " <> sourceFile
               return Nothing
             CabalDidNotMention cabalInfo -> do
-              when debugEnabled $ do
-                relativeCabalFile <-
-                  makeRelativeToCurrentDirectory (ciCabalFilePath cabalInfo)
-                hPutStrLn stderr $
-                  "Found .cabal file "
-                    <> relativeCabalFile
-                    <> ", but it did not mention "
-                    <> sourceFile
+              relativeCabalFile <- makeRelativeToCurrentDirectory (ciCabalFilePath cabalInfo)
+              logDebugM "CABAL FILE" $
+                "Found .cabal file "
+                  <> relativeCabalFile
+                  <> ", but it did not mention "
+                  <> sourceFile
```

Comment:
This is technically different behavior than the previous version: if debug is not enabled, the new version still relativizes the cabal file path, even though it's not printed.

I could create a version of `logDebugM` that takes a to-be-computed `IO String` message, but I don't think adding such a bespoke version is worth this relatively cheap call.